### PR TITLE
feat(puppet): add dns record A and cname for puppet.jenkins.io

### PR DESCRIPTION
**Pull request by smerle33**
add A and CNAME dns records as code.

### CODE REVIEW

File: puppet.jenkins.io.tf [hunk 1]
```diff
@@ -113,7 +113,7 @@ resource "azurerm_linux_virtual_machine" "puppet_jenkins_io" {
 }
 
 resource "azurerm_dns_a_record" "azure_puppet_jenkins_io" {
-  name                = "azure.puppet.jenkins.io."
+  name                = "azure.puppet" #azure.puppet.jenkins.io --> A
```
Comment (lemeurherve) [inline puppet.jenkins.io.tf:116]:
```suggestion
  name                = "azure.puppet"
```
Implicit, redundant with the resource type

File: puppet.jenkins.io.tf [hunk 2]
```diff
@@ -122,7 +122,7 @@ resource "azurerm_dns_a_record" "azure_puppet_jenkins_io" {
 }
 
 resource "azurerm_dns_cname_record" "jenkinsio_target_puppet_jenkins_io" {
-  name                = "puppet.jenkins.io"
+  name                = "puppet" #puppet.jenkins.io --> CNAME vers azure.puppet.jenkins.io
```
Comment (lemeurherve) [inline puppet.jenkins.io.tf:125]:
```suggestion
  name                = "puppet"
```
Same as above

File: puppet.jenkins.io.tf [hunk 3]
```diff
@@ -111,3 +111,21 @@ resource "azurerm_linux_virtual_machine" "puppet_jenkins_io" {
...
+  ttl                 = 300
+  records             = [azurerm_public_ip.puppet_jenkins_io.ip_address]
+  tags                = local.default_tags
+}
+
+resource "azurerm_dns_cname_record" "jenkinsio_target_puppet_jenkins_io" {
+  name                = "puppet" #puppet.jenkins.io --> CNAME vers azure.puppet.jenkins.io
+  zone_name           = data.azurerm_dns_zone.jenkinsio.name
+  resource_group_name = data.azurerm_resource_group.proddns_jenkinsio.name
+  ttl                 = 60
```
Comment (lemeurherve) [inline puppet.jenkins.io.tf:128]: Why this TTL is set to 60 while the other to 300?
Because they are records of different types?

**Review by dduportal — APPROVED** [review]
LGTM!

I'm deleting the existing records (manually managed) as the service is already down

File: puppet.jenkins.io.tf (hunk 3 above)
Comment (lemeurherve) [inline puppet.jenkins.io.tf:128]: Why two different TTLs?

Comment (dduportal) [reply puppet.jenkins.io.tf:128]: I guess it's a mistake, gotta fix it in #377 

Good catch

**Pull request merged**